segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/AsyncTableEntryReader.java

```diff
@@ -241,18 +241,18 @@ protected boolean processReadData(ByteArraySegment readData) {
             assert header != null : "acceptResult called with no header loaded.";
 
             // The key has been read.
-            if (!this.keyValidated) {
-                if (header.getKeyLength() != this.soughtKey.getLength()) {
-                    // Key length mismatch. This is not the Table Entry we're looking for.
-                    complete(null);
-                    return false;
-                }
+            if (this.soughtKey != null && header.getKeyLength() != this.soughtKey.getLength()) {
```

Comment:
I made that change because in this PR I am introducing the ability to lookup (and match) all the keys in a `TableBucket`. If you look at `TableBucketReader#124` you'll see that this is invoked using a `null` key. 

A `null` key indicates it will match any key for whatever `TableEntry` it is pointed at. This is described in the Javadoc for the `readEntry` method (check out line 75).

Reply:
That's fine, but I'm concerned about the possibility of an NPE somewhere from line 258 onwards. For example, if `soughtKey == null`, then `this.soughtKey.getLength()` in line 258 will throw.

Reply:
If you check the constructor, it sets `keyValidated = soughtKey == null`. As such, lines 256-267 will never execute in this case, as any key is considered valid in this scenario.

Lines 271 and 288 already handle this in `getKeyData`.